head/src/main/kotlin/io/qalipsis/core/head/report/DataSeriesServiceImpl.kt

```diff
@@ -196,4 +203,27 @@ internal class DataSeriesServiceImpl(
         )
     }
 
+    override suspend fun refresh() {
+        orchestrationCoroutineScope.launch {
+            val dataSeriesEntities = dataSeriesRepository.findAll()
+            val size = dataSeriesEntities.count()
+            logger.trace { "Updating the queries of $size retrieved data series" }
+            dataSeriesEntities.collect { dataSeriesEntity ->
+                val tenant = tenantRepository.findReferenceById(dataSeriesEntity.tenantId)
+                dataSeriesEntity.query = dataProvider.createQuery(
+                    tenant, dataSeriesEntity.dataType, QueryDescription(
+                        filters = convertFiltersEntity(dataSeriesEntity.filters, dataSeriesEntity.valueName),
+                        fieldName = dataSeriesEntity.fieldName,
+                        aggregationOperation = dataSeriesEntity.aggregationOperation,
+                        timeframeUnit = dataSeriesEntity.timeframeUnitAsDuration
+                    )
+                ).takeUnless(String::isNullOrBlank)
+                dataSeriesRepository.save(dataSeriesEntity)
```

Comment:
If possible, a saveAll using a batch would be better. Do you remember why?

Reply:
If we have a large number of data series we don't want to overwhelm the system by calling a save all a once